client/app.jsx

```diff
@@ -38,10 +41,23 @@ export default class App extends React.Component {
       },
       body: JSON.stringify({ name: collectionName })
     };
-    fetch('/api/collection', init)
-      .then(response => response.json())
-      .then(data => {
-      });
+    fetch('/api/collection', init);
+  }
+
+  handleSave(collectionId, businessId) {
+    const init = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ collectionId: collectionId, businessId: businessId })
+    };
+    fetch(('/api/restaurant'), init);
+    const businessIds = this.state.businessIds;
+    businessIds.push(businessId);
+    this.setState({
+      businessIds: businessIds
+    });
```

Comment:
Two points here:
- you are mutating the array in state, use `concat()` to create a copy that includes the new business id. 
- you should do the state update when the `fetch()` promise becomes fulfilled (`.then()`)